CONTRIBUTING.md

```diff
@@ -0,0 +1,157 @@
+For example you could write words as bold (`**bold**`), or in italics (`*italics*`),
+or as a [link][rick_roll] (`[link](https://https://youtu.be/dQw4w9WgXcQ)`) to another webpage.
+
+GitHub has a really helpful page for getting started with
+[writing and formatting Markdown on GitHub][writing_formatting_github].
+
+
+## Understanding issues
+
+Every project on GitHub uses [issues][link_issues] slightly differently.
+
+The following outlines how the ``geomstats`` developers think about these tools.
+
+* **Issues** are individual pieces of work that need to be completed to move the project forwards.
+A general guideline: if you find yourself tempted to write a great big issue that
+is difficult to describe as one unit of work, please consider splitting it into two or more issues.
+
+* Issues are assigned [labels](#issue-labels) which explain how they relate to the overall project's
+goals and immediate next steps.
+
+
+## Making a change
+
+We appreciate all contributions to ``geomstats``,
+but those accepted fastest will follow a workflow similar to the following:
+
+**1. Comment on an existing issue or open a new issue referencing your addition.**
+
+This allows other members of the ``geomstats`` development team to confirm that you aren't
+overlapping with work that's currently underway and that everyone is on the same page
+with the goal of the work you're going to carry out.
+
+[This blog][link_pushpullblog] is a nice explanation of why putting this work in up front
+is so useful to everyone involved.
+
+**2. [Fork][link_fork] the [geomstats repository][link_geomstats] to your profile.**
+
+This is now your own unique copy of ``geomstats``.
+Changes here won't effect anyone else's work, so it's a safe space to explore edits to the code!
+
+Make sure to [keep your fork up to date][link_updateupstreamwiki] with the master repository.
+
+**3. Make the changes you've discussed, following the [geomstats coding style guide](#geomstats-coding-style-guide).**
+
+Try to keep the changes focused.
+If you feel tempted to "branch out" then please make a [new branch][link_branches].
+
+**4. Submit a [pull request][link_pullrequest].**
+
+A member of the development team will review your changes to confirm
+that they can be merged into the main code base.
+
+Pull requests titles should begin with a descriptive prefix:
+
+* ``ENH``: enhancements or new features
+* ``FIX``: bug fixes
+* ``TST``: new or updated tests
+* ``DOC``: new or updated documentation
+* ``STY``: style changes
+* ``REF``: refactoring existing code
+* ``CI``: updates to continous integration infrastructure
+* ``MAINT``: general maintenance
+
+For example: `[ENH] Implementation of Grassmanians`
+
+For works-in-progress, add the ``WIP`` tag in addition to the descriptive prefix.
+Pull-requests tagged with ``[WIP]`` will not be merged until the tag is removed.
+
+## geomstats coding style guide
+
+Ensure that your code is compliant with [PEP8][link_pep8],
```

Comment:
I don't think we are PEP8 compliant - pep8 is super super aggressive. I think we are compliant to a subset.

side note - It would be nice to have something that does these checks automatically and doens't allow someone to submit a PR w/o a clean flake8 report